src/Document.php

```diff
@@ -866,6 +881,23 @@ private function canViewFileFromItilObject($itemtype, $items_id)
         return $result['cpt'] > 0;
     }
 
+    public static function canViewFileFromOtherObject($itemtype, $items_id)
```

Comment:
Maybe, before a check on `Document_Item` should be done first, using following criteria:
```
'WHERE' => [
    'glpi_documents_items.itemtype' => $itemtype,
    'glpi_documents_items.items_id' => $items_id,
]
```